File: c++/medium/3191.minimum_operations_binary_array_equal_1/minimum_flips_binary_array_all_equal_1.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int minOperations(vector<int>& nums) {
        int flipCount = 0;
        int onesCount = 0;

        for(int i = 0; i < nums.size()-2; i++) {
            int left = nums[i];
            int middle = nums[i+1];
            int right = nums[i+2];

            onesCount++;

            if(left == 1) {
                continue;
            }

            nums[i] = left^1;
            nums[i+1] = middle^1;
            nums[i+2] = right^1;
            flipCount++;
        }

        onesCount += nums[nums.size()-2];
        onesCount += nums[nums.size()-1];

        return onesCount == nums.size() ? flipCount : -1;
    }
};

File: python/medium/3191.minimum_operations_binary_array_equal_1/minimum_flips_binary_array_all_equal_1.py
from typing import List

class Solution:
    def minOperations(self, nums: List[int]) -> int:
        flip_count = 0

        for i in range(len(nums)-2):
            left = nums[i]
            middle = nums[i+1]
            right = nums[i+2]

            if left == 0:
                nums[i] = int(not left)
                nums[i+1] = int(not middle)
                nums[i+2] = int(not right)
                flip_count+=1

        if nums[len(nums)-2] != 1 or nums[len(nums)-1] != 1:
            flip_count = -1

        return flip_count
